src/lib/logger.js

```diff
@@ -20,61 +20,27 @@
 */
 
 import { Container } from 'constitute';
-
-import chalk from 'chalk';
-import settings from '../settings';
-
-function isObject(obj: any): boolean {
-  return obj === Object(obj);
-}
-
-function _transform(...params: Array<any>): Array<any> {
-  return params.map((param: any): string => {
-    if (!isObject(param)) {
-      return param;
-    }
-
-    return JSON.stringify(param);
-  });
-}
-
-function getDate(): string {
-  return (new Date()).toISOString();
-}
-
-class Logger {
-  static container: Container;
-
-  static log(...params: Array<any>) {
-    if (settings.SHOW_VERBOSE_DEVICE_LOGS) {
-      Logger._log(`[${getDate()}]`, _transform(...params));
-    }
-  }
-
-  static info(...params: Array<any>) {
-    Logger._log(
-      `[${getDate()}]`,
-      chalk.cyan(_transform(...params)),
-    );
-  }
-
-  static warn(...params: Array<any>) {
-    Logger._log(
-      `[${getDate()}]`,
-      chalk.yellow(_transform(...params)),
-    );
-  }
-
-  static error(...params: Array<any>) {
-    Logger._log(
-      `[${getDate()}]`,
-      chalk.red(_transform(...params)),
-    );
-  }
-
-  static _log(...params: Array<any>): Function {
-    return Logger.container.constitute('LOGGING_FUNCTION')(...params);
-  }
-}
-
-export default Logger;
+import { Logger as DefaultLogger } from './defaultLogger';
+import { ILogger } from '../types';
+
+
+let logger : ILogger = DefaultLogger;
+
+
+export default {
+  error(...params: Array<any>) {
+    logger.error(...params);
+  },
+  info(...params: Array<any>) {
+    logger.info(...params);
+  },
+  initialize(container: Container) {
```

Comment:
I can do this, but in this case you cant override the Logger in main.js, cause this can only be done after defaultBindings was called ???
As far as i understand the binding concept, there should be no Initialization in DefaultBinding at all ???